test/hotspot/jtreg/runtime/cds/appcds/dynamicArchive/PrintSharedArchiveAndExit.java

```diff
@@ -0,0 +1,98 @@
+ *                 jdk.test.lib.classloader.ClassUnloadCommon
+ *                 jdk.test.lib.classloader.ClassUnloadCommon$1
+ *                 jdk.test.lib.classloader.ClassUnloadCommon$TestFailure
+ * @run driver jdk.test.lib.helpers.ClassFileInstaller -jar hello_custom.jar CustomLoadee
+ * @run driver jdk.test.lib.helpers.ClassFileInstaller -jar WhiteBox.jar sun.hotspot.WhiteBox
+ * @run main/othervm -XX:+UnlockDiagnosticVMOptions -XX:+WhiteBoxAPI -Xbootclasspath/a:./WhiteBox.jar PrintSharedArchiveAndExit
+ */
+
+import java.io.File;
+import jdk.test.lib.cds.CDSTestUtils;
+import jdk.test.lib.process.OutputAnalyzer;
+import jdk.test.lib.helpers.ClassFileInstaller;
+
+public class PrintSharedArchiveAndExit extends DynamicArchiveTestBase {
+    private static final String ARCHIVE_NAME = CDSTestUtils.getOutputFileName("top.jsa");
+
+    public static void main(String... args) throws Exception {
+        runTest(PrintSharedArchiveAndExit::testPrtNExit);
+    }
+
+    public static void testPrtNExit() throws Exception {
+        String wbJar = ClassFileInstaller.getJarPath("WhiteBox.jar");
+        String use_whitebox_jar = "-Xbootclasspath/a:" + wbJar;
+        String appJar = ClassFileInstaller.getJarPath("hello.jar");
+        String customJarPath = ClassFileInstaller.getJarPath("hello_custom.jar");
+        String mainAppClass = "HelloUnload";
+
+        dump(ARCHIVE_NAME,
+            use_whitebox_jar,
+            "-XX:+UnlockDiagnosticVMOptions",
+            "-XX:+WhiteBoxAPI",
+            "-Xlog:cds",
+            "-Xlog:cds+dynamic=debug",
+            "-cp", appJar,
+            mainAppClass, customJarPath, "false", "false")
+            .assertNormalExit(output -> {
+                output.shouldContain("Written dynamic archive 0x")
+                      .shouldNotContain("klasses.*=.*CustomLoadee")
+                      .shouldHaveExitValue(0);
+                });
+
+        run(ARCHIVE_NAME,
+            use_whitebox_jar,
+            "-XX:+UnlockDiagnosticVMOptions",
+            "-XX:+WhiteBoxAPI",
+            "-Xlog:class+load",
+            "-Xlog:cds=debug",
+            "-Xlog:cds+dynamic=info",
+            "-cp", appJar,
+            "-XX:+PrintSharedArchiveAndExit",
+            mainAppClass, customJarPath, "false", "true")
+            .assertNormalExit(output -> {
+                output.shouldHaveExitValue(0)
+                      .shouldMatch("Base archive name: .*.jsa")  // given name ends with .jsa, maynot default name.
+                      .shouldMatch("Dynamic archive name: .*" + ARCHIVE_NAME)
+                      .shouldMatch("Base archive version \\d+")
+                      .shouldContain("java.lang.Object boot_loader")
+                      .shouldContain("HelloUnload app_loader")
+                      .shouldContain("CustomLoadee unregistered_loader")
+                      .shouldContain("Shared Builtin Dictionary")
+                      .shouldContain("Shared Unregistered Dictionary")
```

Comment:
Since the print output includes "VM version", perhaps the above should look for it?

Reply:
Thanks, will add the check.